sphinx/util/inspect.py

```diff
@@ -231,3 +231,228 @@ def is_builtin_class_method(obj, attr_name):
     if not hasattr(builtins, safe_getattr(cls, '__name__', '')):  # type: ignore
         return False
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls  # type: ignore
+
+
+class Parameter(object):
+    """Fake parameter class for python2."""
+    POSITIONAL_ONLY = 0
+    POSITIONAL_OR_KEYWORD = 1
+    VAR_POSITIONAL = 2
+    KEYWORD_ONLY = 3
+    VAR_KEYWORD = 4
+    empty = object()
+
+    def __init__(self, name, kind=POSITIONAL_OR_KEYWORD, default=empty):
+        # type: (str, int, Any) -> None
+        self.name = name
+        self.kind = kind
+        self.default = default
+        self.annotation = self.empty
+
+
+class Signature(object):
```

Comment:
Where is the best place for this class? `sphinx.util.inspect`? `sphinx.ext.autodoc.utils`?